std-bits/base/src/main/java/org/enso/base/statistics/Regression.java

```diff
@@ -0,0 +1,58 @@
+package org.enso.base.statistics;
+
+public class Regression {
+  /**
+   * Performs a least squares fit of a line to the data.
+   *
+   * @param known_xs Set of known X values.
+   * @param known_ys Set of known Y values.
+   * @return A fitted linear model (y = Intercept + Slope x) and the r-squared value.
+   * @throws IllegalArgumentException if the number of elements in the arrays is different or a
+   *     singular X value is provided.
+   */
+  public static LinearModel fit_linear(Double[] known_xs, Double[] known_ys)
+      throws IllegalArgumentException, FitError {
+    CorrelationStatistics stats = CorrelationStatistics.compute(known_xs, known_ys);
+
+    double denominator = denominator(stats);
+    if (denominator == 0) {
+      throw new FitError("Singular X value.");
+    }
+
+    double slope = slope(stats, denominator);
+    return new LinearModel(slope, intercept(stats, slope), stats.rSquared());
+  }
+
+  /**
+   * Performs a least squares fit of a line to the data with a given intercept.
+   *
+   * @param known_xs Set of known X values.
+   * @param known_ys Set of known Y values.
+   * @param intercept The intercept of the line.
+   * @return A fitted linear model (y = Intercept + Slope x) and the r-squared value.
+   * @throws IllegalArgumentException if the number of elements in the arrays is different or a
+   *     singular X value is provided.
+   */
+  public static LinearModel fit_linear(Double[] known_xs, Double[] known_ys, double intercept)
+      throws IllegalArgumentException {
+    CorrelationStatistics stats = CorrelationStatistics.compute(known_xs, known_ys);
+    return new LinearModel(slopeWithIntercept(stats, intercept), intercept, stats.rSquared());
+  }
+
+  private static double denominator(CorrelationStatistics stats) {
+    return stats.getTotalXX() - Math.pow(stats.getTotalX(), 2.0) / stats.getCount();
```

Comment:
Are you sure that using `Math.pow(x, 2.0)` is preferred over `x*x`? 

Unless `Math.pow` does some constant folding, it feels like the `x*x` operation is much more efficient on the CPU.

This _is_ a micro optimization, especially as this particular computation is done _once_ per a model... so take it with a grain of salt, but I'm just surprised to see this pattern - personally `x*x` just feels more natural to me.